### ISSUE TITLE
Python: cannot typematch `decimal`

### ISSUE THREAD

**Issue by kMutagene** [fable-compiler/Fable#3971]
### Description

It seems like while instantiation of a decimal results in a python variable of type `Decimal`, type matching is incorrectly transpiled into matching against `"<class \'int\'>"`

### Repro code

See [REPL](https://fable.io/repl/#?code=DYUwLgBAhhC8ECYCyBYAUO0kBmEAec6ExEAtlGAMYAWEAFHgFwQD2ARgFYCUEA7gJZhqREgB8IjAPwQAJiEr9ywCAFoAfBACMI4uID6qjQnToADgCd+AOzDYrEAEQBSAIIP6uKFyA&html=Q&css=Q)

or transpile this code:

```fsharp
let a = 2M

let f x =
    match (x: obj) with
    | :? decimal -> 1
    | _ -> 2

printfn "%A" (f a)
```

resulting python code is this, and will incorrectly print the result `2`:

```python
from decimal import Decimal
from typing import Any
from fable_modules.fable_library.decimal_ import from_parts
from fable_modules.fable_library.string_ import (to_console, printf)
from fable_modules.fable_library.types import uint8

a: Decimal = from_parts(2, 0, 0, False, uint8(0))

def f(x: Any=None) -> int:
    if str(type(x)) == "<class \'int\'>":
        return 1
    else: 
        return 2

arg: int = f(a) or 0
to_console(printf("%A"))(arg)
```

### Expected and actual results

Expected: Should match against the Decimal class which seems to exist (see typeannotation in above code)

Actual: Matches against int and therefore does not catch an actual decimal with the match case

### Related information

* Fable version: 4.24.0

**Comment by MangelMaxime** [comment]
This is available in Fable 5.0.0-alpha.3
